lib/credentials.js

```diff
@@ -0,0 +1,200 @@
+const middleware = require("./middleware");
+const utils = require("./utils");
+
+class Credentials {
+  constructor(jenkins) {
+    this.jenkins = jenkins;
+  }
+
+  /**
+   * Get or update config
+   */
+  async config(id, folderPath, store, domain, xml, opts) {
```

Comment:
From an API perspective I think it would make more sense to have separate `get` and `update` methods?

If someone adds typescript definitions for this in the future it's going to be annoying to deal with the get case always returning an optional config value.

Reply:
I tried to keep it consistent with all other functions like job and view, that used config both for `get` and `update`.
If you want, I can change it.

Reply:
Make sense, we can leave it as-is.